core-bundle/contao/drivers/DC_Table.php

```diff
@@ -3802,10 +3802,11 @@ protected function treeView()
 		if (!empty($topMostRootIds))
 		{
 			static::preloadCurrentRecords($topMostRootIds, $table);
+			$clipboardManager = System::getContainer()->get('contao.data_container.clipboard_manager');
 
 			for ($i=0, $c=\count($topMostRootIds); $i<$c; $i++)
 			{
-				$tree .= $this->generateTree($table, $topMostRootIds[$i], array('p'=>($topMostRootIds[$i - 1] ?? null), 'n'=>($topMostRootIds[$i + 1] ?? null)), $blnHasSorting, -16, $blnClipboard ? $arrClipboard : false, ($GLOBALS['TL_DCA'][$this->strTable]['list']['sorting']['mode'] ?? null) == self::MODE_TREE && $blnClipboard && $topMostRootIds[$i] == $arrClipboard['id'], false, false, $arrFound);
+				$tree .= $this->generateTree($table, $topMostRootIds[$i], array('p'=>($topMostRootIds[$i - 1] ?? null), 'n'=>($topMostRootIds[$i + 1] ?? null)), $blnHasSorting, -16, $blnClipboard ? $arrClipboard : false, $clipboardManager->isCircularReference($table, $topMostRootIds[$i]), false, false, $arrFound);
```

Comment:
The original line uses `$this->strTable` but you use `$table` now. Is this intended?

Reply:
does not matter here. `$table = $this->strTable;` is the first line of this method. I would keep it like this since we use `$table` everywhere else.

Reply:
I was wrong, it does matter actually. Fixed in 911f80111916be5444a20fbc19ddc5e3c3a3f11d